src/rebar3_hex_docs.erl

```diff
@@ -240,13 +240,17 @@ gen_docs(State, Prv) ->
 
 -spec assert_doc_dir(string()) -> true.
 assert_doc_dir(DocDir) ->
-    filelib:is_dir(DocDir) orelse
-        rebar_api:abort( "Docs were not published since they "
+    case rebar_utils:find_files(DocDir, "index.html") of
+        [_] ->
+            true;
+         _ ->
```

Comment:
`rebar_utils:find_files/2` works on a regex and is recursive in its search by default. You could therefore find `["index.html", "subdir/index.html", "subdir2/other-index.html"]` and fail the check because you only look for single-item lists.

Reply:
ahh! Not good. I switched to this function because older versions of OTP don't have `filelib:find_file/2`. Can make a wrapper, but also #213

Reply:
Switched to filelib:is_file/1, this should be solid now I believe.